### PR TITLE
Enable numerical tests for LSTM whose initial_h, initial_c, and P are not specified

### PR DESCRIPTION
**Pull request by imaihal**
In ONNX LSTM, `initial_h`, `initial_c`, and `P` are optional, but numerical tests didn't test the cases where these parameters are not specified. They always uses tensors with random values. This PR enables the tests with none value for them.

I need these tests for LSTM for NNPA. LSTM op in the zDNN library used in NNPA only supports LSTM whose weight tensor for peepholes is not specified.

Following tests are newly added as an example.
```
  func @main_graph(%arg0: tensor<3x3x2xf32>, %arg1: tensor<1x3x2xf32>, %arg2: tensor<1x\
3x2xf32>) -> (tensor<3x1x3x2xf32>, tensor<1x3x2xf32>, tensor<1x3x2xf32>) {
    %0 = "onnx.NoValue"() {value} : () -> none
   :
    %Y, %Y_h, %Y_c = "onnx.LSTM"(%arg0, %1, %2, %3, %0, %0, %0, %0) {direction = "forwa\
rd", hidden_size = 2 : si64, input_forget = 0 : si64} : (tensor<3x3x2xf32>, tensor<1x8x\
2xf32>, tensor<1x8x2xf32>, tensor<1x16xf32>, none, none, none, none) -> (tensor<3x1x3x2\
xf32>, tensor<1x3x2xf32>, tensor<1x3x2xf32>)
    return %Y, %Y_h, %Y_c : tensor<3x1x3x2xf32>, tensor<1x3x2xf32>, tensor<1x3x2xf32>
  }
```

### CODE REVIEW

File: test/modellib/LSTMModel.cpp [hunk 1]
```diff
@@ -167,8 +186,17 @@ bool LSTMLibBuilder::verifyOutputs() {
...
+  if (isNoneH)
+    initialH =
+        omTensorCreateWithRandomData<float>(llvm::makeArrayRef(hShape), 0, 0);
+  else
+    initialH = omTensorListGetOmtByIndex(inputs, 1);
+  if (isNoneC)
+    initialC =
+        omTensorCreateWithRandomData<float>(llvm::makeArrayRef(cShape), 0, 0);
+  else
+    initialC = omTensorListGetOmtByIndex(inputs, 2);
```
Comment (tungld) [inline test/modellib/LSTMModel.cpp:199]: It's better to move these code into `prepareInputs`: 
```C
list[1] =
      omTensorCreateWithRandomData<float>(llvm::makeArrayRef(hShape), 0.0, (isNoneH) ? 0.0 : 1.0);
list[2] =
      omTensorCreateWithRandomData<float>(llvm::makeArrayRef(cShape), 0.0, (isNoneC) ? 0.0 : 1.0);
```

File: test/numerical/TestLSTM.cpp [hunk 2]
```diff
@@ -89,6 +96,17 @@ int main(int argc, char *argv[]) {
...
+          // Static dimensions (initial_h, initial_c and p are not specified)
+          // forward
+          assert(isOMLSTMTheSameAsNaiveImplFor(
+              1, s, b, i, h, false, false, true, true, true));
+          // reverse
+          assert(isOMLSTMTheSameAsNaiveImplFor(
+              -1, s, b, i, h, false, false, true, true, true));
+          // bidirectional
+          assert(isOMLSTMTheSameAsNaiveImplFor(
+              2, s, b, i, h, false, false, true, true, true));
```
Comment (tungld) [inline test/numerical/TestLSTM.cpp:109]: It looks like we have more and more repeated calls, it is time to use `for` loops for dynamicS, dynamicB, initial_h, initial_c, and p.

**Review by tungld — CHANGES_REQUESTED** [review]

**Review by tungld — APPROVED** [review]
LGTM! Thanks!

**Comment by imaihal** [comment]
@tungld Thanks for reviewing and merging. I thought I let you know after passing the tests :)

**Pull request merged**